valhalla/thor/costmatrix.h

```diff
@@ -157,7 +158,7 @@ class CostMatrix {
   std::vector<EdgeStatus> target_edgestatus_;
 
   // Mark each target edge with a list of target indexes that have reached it
-  std::unordered_map<baldr::GraphId, std::vector<uint32_t>> targets_;
+  robin_hood::unordered_map<baldr::GraphId, std::vector<uint32_t>> targets_;
```

Comment:
Just for my own understanding, what sort of access pattern does this hashmap see? Do we do a lot of inserts & deletes, or is it mostly a bunch of inserts in the beginning & then lots of lookups later?

Reply:
@mandeepsandhu it should be all lookups

Reply:
@mandeepsandhu this map is:

  1. It's used to track of the reverse search has connected with a forward search: https://github.com/valhalla/valhalla/blob/master/src/thor/costmatrix.cc#L397-L401
  2. Each time a node is expanded, an entry is added: https://github.com/valhalla/valhalla/blob/master/src/thor/costmatrix.cc#L618 and https://github.com/valhalla/valhalla/blob/master/src/thor/costmatrix.cc#L822

I uncovered this by running the code in a profiler and looking for hotspots, not by reading through the algorithm itself.

Reply:
Thanks @kevinkreiser . Thanks for the additional info/context @danpat (what tool did you profile it with?)